squirrel/commands/watch.py

```diff
@@ -72,42 +69,44 @@ def get_daemon_pid() -> int:
 
 
 def daemon(wd, logger):
-    watch_flags = flags.CREATE | flags.MODIFY | flags.DELETE
 
     os.chdir(wd)
-    logger.debug('Adding inotify watches')
-    watches = (wd, )
-    i = INotify()
-    for watch in watches:
-        i.add_watch_recursive(watch, watch_flags)
-
-    engine = plugin.load_module()
-
+    watches = wd
+    project_files = Plugin.get_files(wd)
+    logger.info(f'{Plugin.get_files.__name__}: Project files intialized.')
+    logger.info(
+        f'{Plugin.get_files.__name__}: {len(project_files)} files found')
+    engine = Plugin.load_module()
+
+    # TODO?: Add filetype to project.xml? So it can be project spesific files
+    # can be called from something like "engine.file_type"?
+    file_type = ['*.txt']
+    event_handler = Handler(patterns=file_type)
+    observer = Observer()
+    observer.schedule(event_handler, watches, recursive=True)
+    observer.start()
+    logger.debug('Watchdog initialized')
     while True:
-        events = i.read()
-
-        files = get_files(wd)
-
-        # lazzy fix for when we get event from hidden files
-        # or files from hidden directories
-        count = False
-        for event in events:
-            fullpath = os.path.join(i.get_path(event.wd), event.name)
-            if fullpath in files:
-                count = True
-                break
-
-        if count:
-            # time get_count
+        # If modified file found by Watchdog
+        if event_handler.files:
+            for file in event_handler.files:
+                logger.info(f'Found a modified file {file.split("/")[-1]}')
+                # Check if modified file exists in project_file list
+                if file not in project_files:
+                    project_files.append(file)
```

Comment:
Great this will improve performance as we won't be rerunning `find` command over and over. However, there's a problem with temporary files (`a.txt~`, etc.) that some text editors such as vim create. We don't want those to calculated. I quick fix would be to test if the file exists. And we should also make sure that the file is not is a hidden directory.

Reply:
>However, there's a problem with temporary files (a.txt~, etc.) that some text editors such as vim create. We don't want those to calculated.

Isn´t temporary files mostly hidden? Watchdog is set to ignore  `.*` can add `*~` and other extensions, but there might be a lot of combinations? 
On the other hand this might just be a problem with the ´text´ plugin? As `texcount` is only counting `.tex` and i recon `word` will only count `('.doc', '.docx')`


>A quick fix would be to test if the file exists

We could have a for loop in `text` that check if file exists on every run? If the file exists in the timeframe that the files become appeded to the `project_files` variable it will try to count it every time

```python
def files_exists(files):
    count_files = []
    for file in files:
        if os.path.exists(file):
           count_files.append(files)
   return count_files  
```

>And we should also make sure that the file is not is a hidden directory.

Added this in [Watchdog now ignore hidden folders](https://github.com/squirrel-writer/squirrel/pull/4/commits/202bbe2b6a6976574f7faa8b4b7592dfd65d0e83)

```Python
def not_hidden_folder(self, file):
    """Checks for hidden folders"""
    path = file.split('/')
    for dir in path:
        if dir.startswith('.'):
            return False
            break
    return True

def on_created(self, event):
    """Event is created, you can process it now"""
    # if statement to prevent 'files' to have more than one item of each file
    if self.not_hidden_folder(event.src_path):
        if event.src_path not in self.files:
            self.files.append(event.src_path)
````

Reply:
>Isn´t temporary files mostly hidden? Watchdog is set to ignore .* can add *~ and other extensions, but there might be a lot of combinations?

Yeah, that's why I suggested testing if they exist.

>On the other hand this might just be a problem with the text plugin? As texcount is only counting .tex and i recon word will only count ('.doc', '.docx')

It's true that plugins will be handling file types for now, but they should not however be responsible for verifying if the files exist.

>Added this in Watchdog now ignore hidden folders

Great!